app/fetchers.ts

```diff
@@ -1,23 +1,20 @@
+import { useCallback, useEffect, useState } from 'react';
 import { ViewService } from '@penumbra-zone/protobuf';
-import { createPenumbraClient } from '@penumbra-zone/client/create';
+import { createServiceClient } from '@penumbra-zone/client';
 import { bech32mAddress } from '@penumbra-zone/bech32m/penumbra';
 import { joinLoHiAmount } from '@penumbra-zone/types/amount';
 import { getMetadataFromBalancesResponseOptional, getAmount } from '@penumbra-zone/getters/balances-response';
-import { useCallback, useEffect, useState } from 'react';
-
-const createFetchClient = (wallet: string) => {
-  return createPenumbraClient<typeof ViewService>(ViewService, wallet);
-};
+import { client } from './penumbra';
 
-export const fetchAddress = async (wallet: string, account: number): Promise<string | undefined> => {
-  const client = await createFetchClient(wallet);
-  const res = await client.addressByIndex({ addressIndex: { account } });
+export const fetchAddress = async (account: number): Promise<string | undefined> => {
+  const viewService = createServiceClient(client, ViewService);
```

Comment:
Can we make the service client external to this function? Aka, is the approval side effect fixed?

Reply:
Moved the `viewService` atop the fetchers, works fine 👍